### PR TITLE
[FIX] website: prevent social media tour to fail sometimes

### PR DESCRIPTION
**Pull request by rdeodoo**
There is a weird deeper low level misbehavior which makes concurent click to not acts as they should.
Depending on the runbot/tour speed, the misbehavior might kick in and makes the tour fail.

When you click on multiple options very quick back to back, their click will be registered and processed one by one, waiting for the previous one before considering the next one.
You can see that by simply printing a console log in both `renderListItems` and `_computeWidgetState` method from `s_social_media` `options.js` file. Then click quickly on the options related to this snippet like the active toggle and/or remove custom media.

Despite respecting the click order and not overlapping, some click results (like hidding or removing) will be rollbacked visually and only the latest click result (starting from the DOM state before the first click) will be applied.

Long story short: spam click on every toggle option of all the social media, you will see that all your click will be processed one by one:
- The first media you toggled off will be toggled off
- Then the second media you toggled off will be toggled off but the first one will be back to toggle on.

It seems to be correctly applying the click result one after the other, but always starting from the initial DOM state/option widget state (before the clicks), and not as it should: process the second click based on the state of things altered by the previous click.

This will need a deeper and longer investigation to fix the root cause. In the meantime, as this tour is failing multiple times a day, this commit introduce a workaround to avoid this error in the tour.

runbot-16628

### CODE REVIEW

**Comment by rdeodoo** [comment]
Waiting for multi build to confirm that this is fixing the issue.
The bug probably only happens in master, but in doubt the fix is made when the tour was introduced.
I will make sure that without the fix the multi build will be red on master `master-staging-rde` and that with the fix it's only green.

**Comment by rdeodoo** [comment]
The check on master have been done, the fix has been adapted: it now fixes other steps too which were detected by running the tour locally with `watch=True` which seems to slow down the test so much that other issues appears: basically you can't run the tour at all, it fails at multiple place.
Multi-build is green.

File: addons/website/static/tests/tours/snippet_social_media.js [hunk 1]
```diff
@@ -3,6 +3,20 @@
...
+// When the tour was clicking on the toggle to hide facebook in the next step,
+// it would actually "ignore" the result of the click on the toggle and would
+// just consider the action of focusing out the input.
+const preventRaceConditionStep = {
+    content: "Wait a second to avoid race condition",
+    trigger: 'iframe .s_social_media',
+    async run() {
+        await new Promise(r => setTimeout(r, 1000));
+    }
+};
```
Comment (qsm-odoo) [inline addons/website/static/tests/tours/snippet_social_media.js:18]: This does nothing: `run` functions' returns are just ignored so here you don't delay anything. If this solves a race condition, I guess it is just by chance since you add a tour step which may give a couple extra ms for the next step to occur... but definitely not a second :wink:

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

File: addons/website/static/tests/tours/snippet_social_media.js (hunk 1 above)
Comment (rdeodoo) [reply addons/website/static/tests/tours/snippet_social_media.js]: Indeed, actually just a `0` timeout is enough to fix all those bugs.
Probably something related to the JS single thread execution on which `setTimeout` acts special. It's a bit above my head to figure exactly.
Seems reasonable as is now? Multi build & local `watch=True` are all green :+1:

Comment (qsm-odoo) [reply addons/website/static/tests/tours/snippet_social_media.js]: I don't see how a setTimeout 0 that way would change anything about the execution :grimacing: Are you sure that leaving the run method empty does not do exactly the same?

Comment (rdeodoo) [reply addons/website/static/tests/tours/snippet_social_media.js]: I pushed another way of bypassing the race condition, more verbose but at least straightforward.
Ran 2x multi builds in master and all [green](https://runbot.odoo.com/runbot/batch/983078/build/24633281) still

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

File: addons/website/static/tests/tours/snippet_social_media.js [hunk 2]
```diff
@@ -3,6 +3,37 @@
...
+}, {
+    content: "Remove the race condition class for later call of those steps",
+    trigger: 'iframe .s_social_media',
+    run() {
+        document.querySelector('body').classList.remove(socialRaceConditionClass);
+    }
+}, {
+    content: "Ensure the race condition class is removed",
+    trigger: `body:not(.${socialRaceConditionClass})`,
+    run: () => {}, // This is a check.
```
[the remark this answers is not in the document]
Comment (qsm-odoo) [reply addons/website/static/tests/tours/snippet_social_media.js:34]: @rdeodoo EDIT: actually `document.querySelector('body')` => `document.body` (or in this case, even `this.$anchor` is available)

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

**Review by qsm-odoo — APPROVED** [review]
@robodoo r+

**Comment by rdeodoo** [comment]
Small issue fixed with https://github.com/odoo/odoo/pull/114594
See https://github.com/odoo/odoo/pull/114594#issuecomment-1458294869
